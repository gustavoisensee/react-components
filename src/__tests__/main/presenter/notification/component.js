import React from 'react';
import renderer from 'react-test-renderer';
import PresenterNotification from './../../../../modules/main/presenter/notification/component';

describe('Main/presenter/notification component test', () => {
  test('Main/presenter/notification component renders', () => {
    const props = {
      notification: {
        open: true,
        title: 'Title test',
        message: 'Message test',
        notiType: 1, // 1-success, 2-info, 3-warning, 4-error/fail,
        vertical: 1, // 1-top, 2-middle, 3-bottom
        horizontal: 2 // 1-left, 2-center, 3-right
      },
      toggleNotification: () => {},
      changeVerticalNotification: () => {},
      changeHorizontalNotification: () => {},
      changeTypeNotification: () => {}
    };

    const tree = renderer.create(
      <PresenterNotification {...props} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
