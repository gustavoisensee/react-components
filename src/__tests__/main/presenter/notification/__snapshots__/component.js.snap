exports[`Main/presenter/notification component test Main/presenter/notification component renders 1`] = `
<div
  className="content">
  <div
    className="title">
    <span>
      Notification
    </span>
  </div>
  <div
    className="filters">
    <div
      className="filter__span">
      <span>
        Vertical
      </span>
    </div>
    <div
      className="filter__buttons">
      <div
        className="filter__first">
        <input
          checked={true}
          className="radiobutton"
          id="top"
          name="vertical"
          onChange={[Function]}
          type="radio"
          value={1} />
        <label
          className="radiobutton__label"
          htmlFor="top">
          Top
        </label>
      </div>
      <div
        className="filter">
        <input
          checked={false}
          className="radiobutton"
          id="middle"
          name="vertical"
          onChange={[Function]}
          type="radio"
          value={2} />
        <label
          className="radiobutton__label"
          htmlFor="middle">
          Middle
        </label>
      </div>
      <div
        className="filter__last">
        <input
          checked={false}
          className="radiobutton"
          id="bottom"
          name="vertical"
          onChange={[Function]}
          type="radio"
          value={3} />
        <label
          className="radiobutton__label"
          htmlFor="bottom">
          Bottom
        </label>
      </div>
    </div>
  </div>
  <div
    className="filters">
    <div
      className="filter__span">
      <span>
        Horizontal
      </span>
    </div>
    <div
      className="filter__buttons">
      <div
        className="filter__first">
        <input
          checked={false}
          className="radiobutton"
          id="left"
          name="horizontal"
          onChange={[Function]}
          type="radio"
          value={1} />
        <label
          className="radiobutton__label"
          htmlFor="left">
          Left
        </label>
      </div>
      <div
        className="filter">
        <input
          checked={true}
          className="radiobutton"
          id="center"
          name="horizontal"
          onChange={[Function]}
          type="radio"
          value={2} />
        <label
          className="radiobutton__label"
          htmlFor="center">
          Center
        </label>
      </div>
      <div
        className="filter__last">
        <input
          checked={false}
          className="radiobutton"
          id="right"
          name="horizontal"
          onChange={[Function]}
          type="radio"
          value={3} />
        <label
          className="radiobutton__label"
          htmlFor="right">
          Right
        </label>
      </div>
    </div>
  </div>
  <div
    className="filters">
    <div
      className="filter__span">
      <span>
        Type
      </span>
    </div>
    <div
      className="filter__buttons">
      <div
        className="filter__first">
        <input
          checked={true}
          className="radiobutton"
          id="success"
          name="typeNoti"
          onChange={[Function]}
          type="radio"
          value={1} />
        <label
          className="radiobutton__label"
          htmlFor="success">
          Success
        </label>
      </div>
      <div
        className="filter">
        <input
          checked={false}
          className="radiobutton"
          id="info"
          name="typeNoti"
          onChange={[Function]}
          type="radio"
          value={2} />
        <label
          className="radiobutton__label"
          htmlFor="info">
          Info
        </label>
      </div>
      <div
        className="filter">
        <input
          checked={false}
          className="radiobutton"
          id="warning"
          name="typeNoti"
          onChange={[Function]}
          type="radio"
          value={3} />
        <label
          className="radiobutton__label"
          htmlFor="warning">
          Warning
        </label>
      </div>
      <div
        className="filter__last">
        <input
          checked={false}
          className="radiobutton"
          id="error"
          name="typeNoti"
          onChange={[Function]}
          type="radio"
          value={4} />
        <label
          className="radiobutton__label"
          htmlFor="error">
          Error
        </label>
      </div>
    </div>
  </div>
  <div
    className="buttons">
    <button
      className="btn"
      onClick={[Function]}>
      Notification
    </button>
  </div>
  <div
    className="description">
    Description
  </div>
  <div
    className="notification__container top show">
    <div
      className="notification top center success">
      <div
        className="title">
        <span
          className="title__span">
          Title test
          Â 
        </span>
        <button
          className="title__btn"
          onClick={[Function]}>
          x
        </button>
      </div>
      <div
        className="message">
        <span>
          Message test
        </span>
      </div>
    </div>
  </div>
</div>
`;
